File: daily_python/3488_Closest_Equal_Element_Queries.py
import os
import sys
from typing import List

class Solution:
    def solveQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        n = len(nums)
        q = len(queries)
        dis = [n] * n
        m = {}
        for i in range(n):
            m[nums[i]] = -n + i
        for i in range(n):
            dis[i] = i - m[nums[i]]
            m[nums[i]] = i
        for i in range(n - 1, -1, -1):
            m[nums[i]] = n + i
        for i in range(n - 1, -1, -1):
            dis[i] = min(dis[i], m[nums[i]] - i)
            m[nums[i]] = i
        for i in range(n):
            if dis[i] == n:
                dis[i] = -1
        ans = [dis[q] for q in queries]
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            queries = list(map(int, lines[1].split('=')[1].strip()[1:-2].split(',')))
            dataset.append((nums, queries))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums, queries in dataset:
        results.append(solution.solveQueries(nums, queries))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/3488_Closest_Equal_Element_Queries.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> solveQueries(vector<int>& nums, vector<int>& queries) {
        int n = nums.size();
        unordered_map<int,int> m;
        vector<int> dis(n, INT_MAX);
        for (int i = 0; i < n; i++) m[nums[i]] = -n + i;
        for (int i = 0; i < n; i++){
            dis[i] = i - m[nums[i]];
            m[nums[i]] = i;
        }
        for (int i = n - 1; i >= 0; i--) m[nums[i]] = n + i;
        for (int i = n - 1; i >= 0; i--){
            dis[i] = min(dis[i], m[nums[i]] - i);
            m[nums[i]] = i;
        }
        for (int i = 0; i < n; i++) if (dis[i] == n) dis[i] = -1;
        int q = queries.size();
        vector<int> ans(q);
        for (int i = 0; i < q; i++) ans[i] = dis[queries[i]];
        return ans;
    }
};

int main(){
    vector<int> nums = {1,3,1,4,1,3,2};
    vector<int> queries = {0,3,5};
    // vector<int> nums = {1,2,3,4};
    // vector<int> queries = {0,1,2,3};
    Solution S;
    vector<int> ans = S.solveQueries(nums, queries);
    for (int i : ans) cout << i << " ";
    cout << endl;
    return 0;
}